PBlib.py

```diff
@@ -437,9 +446,224 @@ def write_count_matrix(pb_count, outfile, first=1):
+    `seq1` to `seq2` for each position. Both sequences must have the same
+    length.
+
+    ..note:
+
+        The score to move from or to a Z block (dummy block) is always 0.
+
+    Exceptions
+    ----------
+    InvalidBlockError : encountered an unexpected PB
+    """
+    assert len(seq1) == len(seq2), \
+        "sequences have different sizes:\n{}\nvs\n{}".format(seq1, seq2)
+    score = []
+    for pb1, pb2 in zip(seq1, seq2):
+        # score is 0 for Z (dummy PB)
+        if "z" in [pb1.lower(), pb2.lower()]:
+            score.append(0)
+        elif pb1 in NAMES and pb2 in NAMES:
+            score.append(score_mat[NAMES.index(pb1)][NAMES.index(pb2)])
+        else:
+            invalid = []
+            for pb in (pb1, pb2):
+                if not pb in NAMES:
+                    invalid.append(pb)
+            raise InvalidBlockError(', '.join(invalid))
+    return score
+
+
+def matrix_to_single_digit(matrix):
+    """
+    Convert a similarity substitution matrix to a single digit distance
+    substitution matrix
+
+    This function takes a substitution matrix expressed as similarity scores,
+    and expresses it as integer distances in the range [0; 9]. Where 0 means
+    similar PBs, and 9 means different PBs.
+
+    Such single digit substitution matrix is convenient to display sequences
+    of substitution scores.
+
+    ..note:
+
+        If a substitution matrix expressed with distances is provided
+        rather than a matrix expressed with similarity scores, then the
+        returned matrix is expressed with similarity scores.
+    """
+    mini = numpy.min(matrix)
+    maxi = numpy.max(matrix)
+    # Normalize between 0 and 1
+    mat_modified = (matrix + abs(mini))/(maxi - mini)
+    # Convert similarity scores to distances and change the range to [0; 9]
+    mat_modified = 9 * (1 - mat_modified)
+    # Convert to integers
+    mat_modified = mat_modified.astype(int)
+    # Set diagonal to 0
+    for idx in range(len(mat_modified)):
+        mat_modified[idx, idx] = 0
+    return mat_modified
+
+
+def distance_matrix(sequences, substitution_mat):
+    """
+    Compute distances of all sequences against all the others
+
+    The substitution matrix is expected to be expressed as similarity scores.
+    """
+    distance_mat = numpy.empty((len(sequences), len(sequences)), dtype='float')
+
+    print("Building distance matrix")
+    # Get similarity score
+    for i, seqA in enumerate(sequences):
+        sys.stdout.write("\r%.f%%" % (float(i+1)/len(sequences)*100))
+        sys.stdout.flush()
+        for j, seqB in enumerate(sequences[i:], start=i):
+            score = substitution_score(substitution_mat, seqA, seqB)
+            distance_mat[i, j] = score
+            distance_mat[j, i] = score
+    print("")
+    # Set equal the diagonal
+    diag_mini = numpy.min(distance_mat.diagonal())
+    for i in range(len(sequences)):
+        distance_mat[i, i] = diag_mini
```

Comment:
I agree with you @jbarnoud.
However, I believe I made an error in the normalization formula.
It should be
distance_mat = 1 - (distance_mat - mini)/(maxi - mini)
instead of
distance_mat = 1 - (distance_mat + abs(mini))/(maxi - mini)

Could you open an issue for this please?